File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/enums/CipherTypeEnum.java
package org.javamaster.invocationlab.admin.enums;

import java.util.Objects;

public enum CipherTypeEnum {
    AES("AES"),
    MYSQL_AES("MYSQL-AES"),
    ;

    public final String type;

    CipherTypeEnum(String type) {
        this.type = type;
    }

    public static CipherTypeEnum getByType(String type) {
        for (CipherTypeEnum value : CipherTypeEnum.values()) {
            if (Objects.equals(value.type, type)) {
                return value;
            }
        }
        throw new IllegalArgumentException("wrong type:" + type);
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/enums/CipherTypeEnum.kt
package org.javamaster.invocationlab.admin.enums

enum class CipherTypeEnum(@JvmField val type: String) {
    AES("AES"),
    MYSQL_AES("MYSQL-AES"),
    ;

    companion object {
        @JvmStatic
        fun getByType(type: String): CipherTypeEnum {
            for (value in entries) {
                if (value.type == type) {
                    return value
                }
            }
            throw IllegalArgumentException("wrong type:$type")
        }
    }
}
